### PR TITLE
EAN Validator

### PR DESCRIPTION
**Pull request by angy113**
@carledriss ready to code review

### CODE REVIEW

File: EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Compares if the thirteenth digit of the code is the correct.
+     *
+     * @param number Is the String of thirteen numbers.
+     * @return true if the digit is correct, otherwise returns false.
+     */
+    public static boolean validate(String number) {
+        final int checkSumPosition = 12;
+        final int correctCheckSum = obtainCheckSum(number);
+        return number.charAt(checkSumPosition) - 48 == correctCheckSum;
```
Comment (carledriss) [inline EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java]: why are you using 48 ?
I think you can use Character.getNumericValue() method.

File: EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+     * At the end the new numbers are summed.
+     *
+     * @param number Is the String of thirteen numbers.
+     * @return The total sum of the numbers till the twelve position.
+     */
+    private static int calculateSum(String number) {
+        final int limit = 12;
+        return (int) IntStream.range(1, number.length())
+                .limit(limit)
+                .mapToDouble(index -> index % 2 == 0 ? (number.charAt(index - 1) - 48) * 3 : number.charAt(index - 1) - 48)
```
Comment (carledriss) [inline EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java]: same as above

**Comment by angy113** [comment]
@carledriss I aready replaced the function by Character.getNuricValue()

File: EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+public class EANValidator {
+
+    /**
+     * Compares if the thirteenth digit of the code is the correct.
+     *
+     * @param number Is the String of thirteen numbers.
+     * @return true if the digit is correct, otherwise returns false.
+     */
+    public static boolean validate(String number) {
+        final int checkSumPosition = 12;
```
Comment (carledriss) [inline EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java]: This value is hardcoded
We can define 

``` java
number.length() - 1
```

File: EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+     * If the number is in pair position, it is converted to the
+     * same number multiplied per three.
+     * If the number is in odd position, its value is maintained.
+     * At the end the new numbers are summed.
+     *
+     * @param number Is the String of thirteen numbers.
+     * @return The total sum of the numbers till the twelve position.
+     */
+    private static int calculateSum(String number) {
+        final int limit = 12;
```
Comment (carledriss) [inline EANCode/main/java/org/fundacionjala/EANCode/EANValidator.java]: same as above

**Comment by angy113** [comment]
@carledriss ready to code review

**Comment by carledriss** [comment]
Looks good to me!

**Pull request merged**